test/val/val_id_test.cpp

```diff
@@ -177,9 +185,62 @@ std::string BranchConditionalTail = R"(
                OpFunctionEnd
 )";
 
+std::string ValidateIdWithMessage::make_message(const char* msg) {
+  const bool use_friendly_names = GetParam();
+  if (use_friendly_names) {
+    return msg;
+  }
+
+  std::string message(msg);
+  std::string result;
+
+  size_t next = 0;
+  while (next < message.size()) {
+    // Parse 'num[%name]'
+    size_t open_quote = message.find('\'', next);
+    fprintf(stderr, "%zu\n", open_quote);
+
+    // Copy up to the first quote
+    result.append(message, next, open_quote - next);
```

Comment:
nit:  I normally use a std::ostringstream to accumulate the string parts and then call .str() on it at the end, rather than constantly appending to a std::string.
I don't know if it's any faster.

Reply:
It may also be more readable, using chained << operators.

Reply:
Follow up: https://github.com/KhronosGroup/SPIRV-Tools/pull/4956